src/pages/VenueEntrancePage/VenueEntrancePage.tsx

```diff
@@ -62,14 +66,9 @@ export const VenueEntrancePage: React.FC = () => {
   }
 
   const stepConfig = world.entrance?.[step - 1];
-  if (!stepConfig) {
+  if (Number.isNaN(step) || !stepConfig) {
```

Comment:
This might be another, albeit less likely, suspect for the cause of the heisenbug.